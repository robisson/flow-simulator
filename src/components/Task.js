import React from "react";

export default function Task({ name }) {
  return <li>{name}</li>;
}
