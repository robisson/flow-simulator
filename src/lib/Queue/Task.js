export default class Task {
  constructor(id, name, costOfDelay, estimation, estimationDate, resources) {
    this.id = id;
    this.name = name;
    this.costOfDelay = costOfDelay;
    this.estimation = estimation;
    this.estimationDate = estimationDate;
    this.resources = resources;
  }

  getId() {
    return this.id;
  }

  getName() {
    return this.name;
  }

  getCostOfDelay() {
    return this.costOfDelay;
  }

  getEstimation() {
    return this.estimation;
  }

  setEstimationDate(estimationDate) {
    this.estimationDate = estimationDate;
  }

  getEstimationDate() {
    return this.estimationDate;
  }

  setResources(resources) {
    this.resources = resources;
  }

  getResources() {
    return this.resources;
  }

  isLate(dateToCompare) {
    return this.estimationDate.diff(dateToCompare, "days") < 0;
  }
}
