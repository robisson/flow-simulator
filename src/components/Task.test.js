import React from "react";
import { mount } from "enzyme";
import Task from "./Task";
import "../../setupTests";

describe("<Task /> component", () => {
  it("should render component correctly with name property", () => {
    let wrapper = mount(<Task name="test" />);

    expect(wrapper.props().name).toEqual('test');

    expect(wrapper).toMatchSnapshot();
  });
});
