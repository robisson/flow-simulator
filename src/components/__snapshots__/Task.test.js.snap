// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Task /> component should render component correctly with name property 1`] = `ReactWrapper {}`;
